planner/core/stats.go

```diff
@@ -419,9 +419,13 @@ func (ds *DataSource) generateIndexMergeOrPaths() {
 		}
 		if len(partialPaths) > 1 {
 			possiblePath := ds.buildIndexMergeOrPath(partialPaths, i)
-			if possiblePath != nil {
-				ds.possibleAccessPaths = append(ds.possibleAccessPaths, possiblePath)
+			sel, _, err := ds.tableStats.HistColl.Selectivity(ds.ctx, []expression.Expression{sf}, nil)
+			if err != nil {
+				logutil.BgLogger().Debug("something wrong happened, use the default selectivity", zap.Error(err))
+				sel = SelectionFactor
 			}
+			possiblePath.CountAfterAccess = sel * ds.tableStats.RowCount
```

Comment:
Why is the nil check for `possiblePath` removed?

Reply:
According to the implementation of `buildIndexMergeOrPath()`, `possiblePath` can't be nil.
https://github.com/pingcap/tidb/blob/e6b2fa861a285babec6be57e30d315b85c06026f/planner/core/stats.go#L548-L553